doc/date/calendars.rdoc

```diff
@@ -0,0 +1,78 @@
+== Julian and Gregorian Calendars
+
+The difference between the
+{Julian calendar}[https://en.wikipedia.org/wiki/Julian_calendar]
+and the
+{Gregorian calendar}[https://en.wikipedia.org/wiki/Gregorian_calendar]
+may matter to your program if it uses dates in the interval:
+
+- October 15, 1582.
+- September 14, 1752.
+
+A date outside that interval (including all dates in modern times)
+is the same in both calendars.
+However, a date _within_ that interval will be different
+in the two calendars.
+
+=== Different Calendar, Different \Date
+
+The reason for the difference is this:
+
+- On October 15, 1582, several countries changed
+  from the Julian calendar to the Gregorian calendar;
+  these included Italy, Poland, Portugal, and Spain.
+  Other contries in the Western world retained the Julian calendar.
+- On September 14, 1752, most of the British empire
+  changed from the Julian calendar to the Gregorian calendar.
+
+When your code uses a date in this "gap" interval,
+it will matter whether it considers the switchover date
+to be the earlier date or the later date (or neither).
+
+=== Argument +start+
+
+Certain methods in class \Date handle differences in the
+{Julian and Gregorian calendars}[rdoc-ref:calendars.rdoc@Julian+and+Gregorian+Calendars]
+by accepting an optional argument +start+, whose value may be:
+
+- Date::ITALY (the default): the created date is Julian
+  if before October 15, 1582, Gregorian otherwise:
+
+      d = Date.new(1582, 10, 15)
+      d.prev_day.julian? # => true
+      d.julian?          # => false
+      d.gregorian?       # => true
+
+- Date::ENGLAND: the created date is Julian if before September 14, 1752,
+  Gregorian otherwise:
+
+    d = Date.new(1752, 9, 14, Date::ENGLAND)
+    d.prev_day.julian? # => true
+    d.julian?          # => false
+    d.gregorian?       # => true
+
+- Date::JULIAN: the created date is Julian regardless of its value:
+
+    d = Date.new(1582, 10, 15, Date::JULIAN)
+    d.julian? # => true
+
+- Date::GREGORIAN: the created date is Gregorian regardless of its value:
+
+    d = Date.new(1752, 9, 14, Date::GREGORIAN)
+    d.prev_day.gregorian? # => true
+
+For a \Date object created with Date::JULIAN or Date::GREGORIAN,
+there are implications when the date is incremented or decremented:
+
+  # Julian date, incremented to a Gregorian date.
+  d0 = Date.new(1582, 1, 1, Date::ITALY)  # => #<Date: 1582-01-01>
+  d0.julian?                              # => true
+  d1 = d0.next_year                       # => #<Date: 1583-01-01 >
+  d1.gregorian?                           # => true
+
+  # Gregorian date, decremented to a Julian date.
+  d0 = Date.new(1583, 1, 1, Date::ITALY)  # => #<Date: 1583-01-01)>
+  d0.gregorian?                           # => true
+  d1 = d0.prev_year                       # => #<Date: 1582-01-01>
+  d1.julian?                              # => true
```

Comment:
I don't see how this example highlights any implications when incrementing or decrementing dates, it seems to work exactly as expected. I think it only affects dates around the switchover date. For example:

```ruby
d = Date.new(1582, 10, 15) # => #<Date: 1582-10-15>
d.prev_day # => #<Date: 1582-10-04>

Date.new(1582, 10, 14) # => Date::Error: invalid date
```

Reply:
> I don't see how this example highlights any implications when incrementing or decrementing dates, it seems to work exactly as expected. I think it only affects dates around the switchover date. For example:
> 
> ```ruby
> d = Date.new(1582, 10, 15) # => #<Date: 1582-10-15>
> d.prev_day # => #<Date: 1582-10-04>
> 
> Date.new(1582, 10, 14) # => Date::Error: invalid date
> ```

Works exactly as you and I expect.  Naive user may be surprised that in/de/crementing from one calendar may lead to the other.

Also, I have not illustrated the error you cite (trying to create into the jump);  should we include that?

Reply:
Right, but the documentation above it says

> For a Date object created with Date::JULIAN or Date::GREGORIAN,
> there are implications when the date is incremented or decremented

I see no implications in the examples that follow. The call to `Date#next_year` returns a date with the same month and day in the following year and the call to `Date#prev_year` returns a date with the same month and day in the previous year. Everything works exactly as expected. So I'm a bit confused as to what the examples are trying to show.

Reply:
Very well;  I've removed them.